buildSrc/src/main/groovy/org/elasticsearch/gradle/doc/RestTestsFromSnippetsTask.groovy

```diff
@@ -235,7 +237,13 @@ public class RestTestsFromSnippetsTask extends SnippetsTask {
                 }
             } else {
                 current.println('---')
-                current.println("\"line_$test.start\":")
+                if (test.name != null && test.name.isBlank() == false) {
+                    def num = names.compute(test.name, { key, old -> old == null ? 1 : old + 1 })
```

Comment:
I'm not 100% sure we want to allow this. 
Any reason they shouldn't all be unique so we can just err here if we detect duplicates ?

Reply:
I was unsure as well. I've changed it to exception.

Reply:
If the ids aren't unique then the docs won't build either1